app/components/move_message.py

```diff
@@ -663,171 +630,283 @@ async def discard(
     async def _edit(self, interaction: dc.Interaction, content: str) -> None:
         self._state.last_update = dt.datetime.now(tz=dt.UTC)
         await interaction.response.edit_message(content=UPLOADING, view=None)
-        await _apply_edit_from_thread(self._state.moved_message, self._message, content)
+        await _apply_edit_from_thread(
+            self._message_cog.bot, self._state.moved_message, self._message, content
+        )
         assert isinstance(self._message.channel, dc.Thread)
-        await _remove_edit_thread(
+        await self._message_cog.remove_edit_thread(
             self._message.channel, self._message.author, action="finished editing"
         )
 
 
-@bot.tree.context_menu(name="Move message")
-@dc.app_commands.default_permissions(manage_messages=True)
-@dc.app_commands.guild_only()
-async def move_message(interaction: dc.Interaction, message: dc.Message) -> None:
-    """
-    Adds a context menu item to a message to move it to a different channel. This is
-    used as a moderation tool to make discussion on-topic.
-    """
-    assert not is_dm(interaction.user)
-
-    if not (is_mod(interaction.user) or is_helper(interaction.user)):
-        await interaction.response.send_message(
-            "You do not have permission to move messages.", ephemeral=True
-        )
-        return
-
-    if not message_can_be_moved(message):
-        await interaction.response.send_message(
-            "System messages cannot be moved.",
-            ephemeral=True,
-            view=DeleteInstead(message),
+@final
+class MoveMessage(commands.Cog):
+    def __init__(self, bot: GhosttyBot) -> None:
+        # A dictionary mapping edit thread IDs to their state. An ID is used instead of
+        # the thread itself because there is a thread available already at all places
+        # this is used, and because it isn't obvious how threads are hashed.
+        self.edit_threads: dict[int, ThreadState] = {}
+        # A dictionary mapping edit thread creator IDs to the ID of the thread they are
+        # editing in. IDs are used for the same reason as above.
+        self.edit_thread_creators: dict[int, int] = {}
+
+        self.bot = bot
+
+        self.context_menus = (
+            dc.app_commands.ContextMenu(
+                name="Move Message",
```

Comment:
https://github.com/ghostty-org/discord-bot/blob/2ed923b8678145098c19612666889c975f4d64ab/app/components/move_message.py#L673

```suggestion
                name="Move message",
```